scripts/segmentation/test.py

```diff
@@ -207,37 +217,87 @@ def benchmarking(model, args):
         else:
-            raise RuntimeError("=> no checkpoint found at '{}'" \
-                .format(args.resume))
+            model.hybridize()
+    else:
+        model_prefix = 'deploy_int8' if args.quantized else 'deploy'
+        model = mx.gluon.SymbolBlock.imports('{}-symbol.json'.format(args.model_prefix),
+              ['data'], '{}-0000.params'.format(args.model_prefix))
+        model.hybridize(static_alloc=True, static_shape=True)
 
     print("Successfully loaded %s model" % model_prefix)
     print('Testing model: ', args.resume)
+
+    # benchmark
+    if args.benchmark:
+        print('------benchmarking on %s model------' % model_prefix)
+        benchmarking(model, args)
+        sys.exit()
+
     # image transform
     input_transform = transforms.Compose([
         transforms.ToTensor(),
         transforms.Normalize([.485, .456, .406], [.229, .224, .225]),
     ])
 
-    if not args.benchmark:
-        if '_int8' in model_prefix:
-            test_quantization(model, args, input_transform)
+    if args.calibration or '_int8' in model_prefix:
+        # get dataset
+        if args.eval:
+            testset = get_segmentation_dataset(
+                args.dataset, split='val', mode=args.mode, transform=input_transform)
         else:
-            test(model, args, input_transform)
+            testset = get_segmentation_dataset(
+                args.dataset, split='test', mode=args.mode, transform=input_transform)
+        size = len(testset)
+        batchify_fn = ms_batchify_fn if testset.mode == 'test' else None
+        # get dataloader
+        test_data = gluon.data.DataLoader(
+                testset, args.batch_size, batchify_fn=batchify_fn, last_batch='rollover',
+                shuffle=False, num_workers=args.workers)
+        
+        # calibration
+        if not args.quantized:
+            assert args.eval and args.mode == 'val', "Only val dataset can used for calibration."
+            exclude_sym_layer = []
+            exclude_match_layer = []
+            if args.ngpus > 0:
+                raise ValueError('currently only supports CPU with MKL-DNN backend')
+            model = quantize_net(model, calib_data=test_data, quantized_dtype=args.quantized_dtype, calib_mode=args.calib_mode, 
+                                 exclude_layers=exclude_sym_layer, num_calib_examples=args.batch_size * args.num_calib_batches,
+                                 exclude_layers_match=exclude_match_layer, ctx=args.ctx[0], logger=logger)
+            dir_path = os.path.dirname(os.path.realpath(__file__))
+            dst_dir = os.path.join(dir_path, 'model')
+            if not os.path.isdir(dst_dir):
+                os.mkdir(dst_dir)
+            prefix = os.path.join(dst_dir, model_prefix + '-quantized-' + args.calib_mode)
+            logger.info('Saving quantized model at %s' % dst_dir)
+            model.export(prefix, epoch=0)
+            sys.exit()
```

Comment:
why sys.exit?

Reply:
We assume that users run this script either in `calibration` or `test` mode. That is, users can get quantized models from their own FP32 models in `calibration` mode firstly and then run inference with these saved models in `test` mode.